bin/zopen-build

```diff
@@ -512,6 +519,52 @@ processOptions()
   done
 }
 
+signPaxFile()
+{
+  # Paths to your files
+  PRIVATE_KEY="${ZOPEN_GPG_SECRET_KEY_FILE}"
+  PASSPHRASE_FILE="${GPG_SECRET_KEY_PASSPHRASE}"
+  FILE_TO_SIGN="${paxFileName}"
+  OUTPUT_SIGNATURE="${paxFileName}.asc"
+
+  # Create a temporary directory for GPG keyring
+  TMP_GPG_DIR="$HOME/tmp_gpg"
+  mkdir -p "$TMP_GPG_DIR"
+  chmod a+w "$TMP_GPG_DIR"
+
+  # Import the private key into the specified keyring files
+  printInfo "Importing private key..."
+  gpg --no-default-keyring --keyring "$TMP_GPG_DIR/pubring.kbx" --secret-keyring "$TMP_GPG_DIR/secring.gpg" --batch --yes --import "$PRIVATE_KEY" > /dev/null 2>&1
+
+  # Verify that the key was imported successfully
+  printInfo "Checking if secret key is imported..."
+  gpg --no-default-keyring --keyring "$TMP_GPG_DIR/pubring.kbx" --secret-keyring "$TMP_GPG_DIR/secring.gpg" --list-secret-keys > /dev/null 2>&1
+  if [ $? -ne 0 ]; then
+      printWarning "No secret key found. Signing aborted."
+      rm -rf "$TMP_GPG_DIR"
```

Comment:
If you use $tmp from above, you could enhance cleanupOnExit to remove it on exit